src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2Extensions.cs

```diff
@@ -203,5 +203,31 @@ public static ArraySegment<byte> GetArraySegment(this MemoryPoolIterator2 start,
             start.CopyTo(array, 0, length, out length);
             return new ArraySegment<byte>(array, 0, length);
         }
+
+        public static bool StartsWithIgnoreCase(this MemoryPoolIterator2 start, string prefix, out MemoryPoolIterator2 end, out bool caseMatches)
```

Comment:
:+1: for profiling. @davidfowl Do you have any specific ideas for improving the perf of these PR?